### ISSUE TITLE
Unable to repair and open a PDF file

### ISSUE THREAD

**Issue by nbenitez** [qpdf/qpdf#1103]
The attached PDF file is succesfully repaired and opened by Adobe Reader, Firefox PDF.js.. but `qpdf` fails  with following errors:

```
$ qpdf --qdf --object-streams=disable bug.pdf bug_uncompressed.pdf
WARNING: bug.pdf: file is damaged
WARNING: bug.pdf: can't find startxref
WARNING: bug.pdf: Attempting to reconstruct cross-reference table
qpdf: bug.pdf: unable to find trailer dictionary while recovering damaged file
```

[bug.pdf](https://github.com/qpdf/qpdf/files/13814392/bug.pdf)

**Comment by jberkenbilt** [comment]
The PDF spec says the `%%EOF` marker should be within the last 1024 bytes. This file has it earlier than that. It should be an easy fix to have qpdf discover the end of file properly. I'll try to get this in for 11.7.1. Thanks for reporting.

**Comment by nbenitez** [comment]
Thanks for giving feedback so quick!

**Comment by jberkenbilt** [comment]
While my analysis contained true statements, simply being more liberal about finding EOF caused other problems with appended files in other recovery cases. I think a deeper issue is that qpdf was unable to recover files with xref streams instead of xref tables. I will be pushing a fix up for that momentarily. With the fix, qpdf generates warnings but is able to recover this file and get results identical to if the trailing junk was removed from the file. Recovering broken files is pretty hard -- it's all heuristics, and it's easy for a change to break previous successful recoveries. I try to make sure I have a test file in the test suite for every specific recovery case qpdf handles.

**Comment by jberkenbilt** [comment]
Just released 11.8.0, which includes this fix.

**Comment by nbenitez** [comment]
Thank you very much! 🙂
